app/models/course/level.rb

```diff
@@ -3,8 +3,12 @@ class Course::Level < ActiveRecord::Base
   validates :experience_points_threshold, numericality: { greater_than_or_equal_to: 0 }
 
   belongs_to :course, inverse_of: :levels
-  default_scope { order(:experience_points_threshold) }
-  attr_writer :level_number
+
+  # See below for Level#ranked. The use of ranked as a default scope would result in the
+  # inability to use certain ActiveRecord methods, eg. Course.levels.count.
+  # A simple workaround would be to state a column name, eg. Course.levels.count(:id) or
+  # Course.levels.count('*')
+  default_scope { with_level_number.order(:experience_points_threshold) }
```

Comment:
I think the order by threshold needs to be specified in the with_level_number scope?